code/code/misc/equip.cc

```diff
@@ -1058,8 +1058,8 @@ int TBeing::wear(TObj *o, wearKeyT keyword, TBeing *ch)
               --(*o);
               equipChar(o, HOLD_RIGHT);
               // learn 2h spec as with dual wield
-              if (doesKnowSkill(SKILL_2H_SPEC_DEIKHAN)) {
-                learnFromDoingUnusual(LEARN_UNUSUAL_NORM_LEARN, SKILL_2H_SPEC_DEIKHAN, 60);
+              if (doesKnowSkill(SKILL_2H_SPEC)) {
```

Comment:
I see you updated getSkillNum() but aren't using it anywhere

Reply:
Updated.

Reply:
And just a note - please confirm this update is using the functions correctly.   I might just be missing something here.

Reply:
So I think what you need here is to first get a reference to which version of 2H Spec the character in question uses, by doing something like `auto skill2HSpec = getSkillNum(SKILL_2H_SPEC)`. Then you would pass that result to the `doesKnowSkill` function (`getSkillNum(skill2HSpec)`) instead of passing `SKILL_2H_SPEC`. Or you could just do `doesKnowSkill(getSkillNum(SKILL_2H_SPEC))` if you don't need to reference the skill again anywhere later on in the code.

It's necessary here because with the way you have it coded currently a Deikhan will never hone 2H spec (I think), since for them `doesKnowSkill(SKILL_2H_SPEC)` will always return false as they don't have SKILL_2H_SPEC - they have SKILL_2H_SPEC_DEIKHAN. So `getSkillNum` when called by a Deikhan will return the skill number for SKILL_2H_SPEC_DEIKHAN instead, which will then return `true` for the `doesKnowSkill` call and allow them to hone their version of the skill.

You can tell which skills you need to use `getSkillNum` for by just looking at the code for that function. Sorry if you already knew all this, I just figured it out myself like yesterday so thought I'd share in case it helps.